lib/honeycomb/integrations/rails.rb

```diff
@@ -89,6 +89,18 @@ class Middleware
       include Rack
       include Warden
       include Rails
+
+      def call_with_hook(env, span, &add_field)
+        super
+      rescue => e
+        wrapped = ActionDispatch::ExceptionWrapper.new(nil, e)
+
+        span.add_field "response.status_code", wrapped.status_code
+
+        raise e
+      ensure
+        add_package_information(env, &add_field)
```

Comment:
maybe this should be a block passed into `call_with_hook`? Then we won't need to replicate the remainder of the above method

Reply:
🤔 Hm. I'm not entirely sure I follow, can you elaborate a little?

Reply:
presumably this ensure block is so that in the event of an exception that this information that is usually added on line 53 of the rack middleware will still be added? If so maybe `call_with_hook` should take a block and call that instead? Small example below

```
status, headers, body = call_with_hook(env, span) do
  add_package_information(env, &add_field)
  ... other things
end


def call_with_hook(env, span)
  super
  rescue => e
  raise e
  ensure
    yield
  end
end
```

Reply:
Ah okay, I think I follow now. The `ensure` block is there because adding that info seemed useful regardless of errors being raised. I think doing this in the `Rack` middleware may accomplish the same thing in a more generalized way...

```ruby
begin
  status, headers, body = call_with_hook(env, span, &add_field)
  span.add_field("response.status_code", status)
  span.add_field("response.content_type", headers["Content-Type"])
ensure
  add_package_information(env, &add_field)
  extract_user_information(env, &add_field)
end
```

This avoid the confusion of a thing like `call_with_hook` possibly taking two blocks and/or getting a bit convoluted about when the block is run. The above could also be added to the `Rack#call_with_hook` method rather than adding to `Rack#call` if that seems more fitting. I personally prefer it in `Rack#call` as it's more apparent what's being done. I wouldn't want the base `Rack#call_with_hook` to do anything so keeping those method calls in the main body feels clearer.

Reply:
@lirossarvet even better!

Reply:
https://github.com/honeycombio/beeline-ruby/pull/153/commits/c512aa1d41ce56d6c6023d5775419865dc9d3eef @martin308 here ya go!

Reply:
Tested the `rails_52.gemfile` and `rails_6.gemfile` and they both passed locally at least. Not exhaustive but promising!

Reply:
@lirossarvet great! If you fix the couple o lint errors then the CI should be able to run all the tests

Reply:
https://github.com/honeycombio/beeline-ruby/pull/153/commits/ec25a14fa6fb7ba1792e29e2f8bd8fe5a7ee4aae

Done! @martin308 thanks so much for the feedback